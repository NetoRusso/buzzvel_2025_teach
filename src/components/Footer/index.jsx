
import styles from "./styles.module.css";
import Image from "next/image";

import logo from "@/assets/SVG/Logo-white.svg";
import arrow from "@/assets/SVG/arrow_right_white.svg";
import wolrd from "@/assets/SVG/world.svg";
import euro from "@/assets/SVG/euro.svg";
import accessibility from "@/assets/SVG/accessibility.svg";



const Footer = () => {
  return (
    <footer className={styles.container}>
      <div className={styles.content}>
        <Image src={logo} alt="logo" width={173} height={35} className={styles.logo} loading="lazy" />
        <div className={styles.box}>
          <div className={styles.box_links}>
            <p className={styles.box_links_item}>
              Product
            </p>
            <p className={styles.box_links_item}>
              Pricing
            </p>
            <p className={styles.box_links_item}>
              Overview
            </p>
            <p className={styles.box_links_item}>
              Browse
            </p>
            <p className={styles.box_links_item}>
              Accessibility <span>BETA</span>
            </p>
          </div>

          <div className={styles.box_links}>
            <p className={styles.box_links_item}>
              Solutions
            </p>
            <p className={styles.box_links_item}>
              Brainstorming
            </p>
            <p className={styles.box_links_item}>
              Ideation
            </p>
            <p className={styles.box_links_item}>
              Wireframing
            </p>
            <p className={styles.box_links_item}>
              Research
            </p>
          </div>

          <div className={styles.box_links}>
            <p className={styles.box_links_item}>
              Resources
            </p>
            <p className={styles.box_links_item}>
              Help Center
            </p>
            <p className={styles.box_links_item}>
              Blog
            </p>
            <p className={styles.box_links_item}>
              Tutorials
            </p>
            <p className={styles.box_links_item}>
              FAQs
            </p>
          </div>

          <div className={styles.box_links}>
            <p className={styles.box_links_item}>
              Support
            </p>
            <p className={styles.box_links_item}>
              Contact Us
            </p>
            <p className={styles.box_links_item}>
              Developers
            </p>
            <p className={styles.box_links_item}>
              Documentation
            </p>
            <p className={styles.box_links_item}>
              Integrations
            </p>
          </div>

          <div className={styles.box_links}>
            <p className={styles.box_links_item}>
              Company
            </p>
            <p className={styles.box_links_item}>
              About
            </p>
            <p className={styles.box_links_item}>
              Press
            </p>
            <p className={styles.box_links_item}>
              Events
            </p>
            <p className={styles.box_links_item}>
              Request Demo <Image src={arrow} alt="arrow" width={24} height={24} loading="lazy" />
            </p>
          </div>
        </div>

      </div>
      <div className={styles.copyright}>
        <p className={styles.copyright_text}>
          uteach &copy; 2023. All rights reserved.
        </p>
        <div className={styles.links_acess}>
          <p className={styles.links_acess_item}>
            Terms
          </p>
          <p className={styles.links_acess_item}>
            Privacy
          </p>
          <p className={styles.links_acess_item}>
            Contact
          </p>
          <p className={styles.links_acess_item}>
            <Image src={wolrd} alt="world" width={24} height={24} loading="lazy" />
            EN
          </p>
          <p className={styles.links_acess_item}>
            <Image src={euro} alt="euro" width={24} height={24} loading="lazy" />
            EUR
          </p>
          <p className={styles.links_acess_item}>
            <Image src={accessibility} alt="accessibility" width={24} height={24} loading="lazy" />
          </p>
        </div>
      </div>
    </footer>
  )
};

export default Footer;
