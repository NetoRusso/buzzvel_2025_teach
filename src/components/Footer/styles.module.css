.container {
  width: 100%;
  height: auto;
  min-height: 420px;
  background-color: var(--blue-Gray-900);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 48px 16px 30px;
}

.content {
  width: 100%;
  max-width: 1280px;
  height: auto;
  border-bottom: 1px solid var(--blue-Gray-700);
  display: flex;
  align-items: flex-start;
  gap: 40px;
  padding-bottom: 40px;
  margin-bottom: 24px;
}

.logo {
  align-self: flex-start;
  flex-shrink: 0;
  margin-top: 0;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
}

.logo:hover {
  scale: 0.95;
}

.box {
  width: 100%;
  flex-grow: 1;
  display: flex;
  flex-flow: row wrap;
  align-items: flex-start;
  gap: 30px 20px;
}

.box_links {
  width: 173px;
  height: auto;
  display: flex;
  flex-flow: column nowrap;
  align-items: flex-start;
  gap: 10px;
}

.box_links_item {
  display: flex;
  align-items: center;
  gap: 8px;
  color: var(--background-white);
  font-size: 16px;
  font-weight: 500;
  line-height: 1.4;
  cursor: pointer;
  transition: color 0.2s ease-in-out;
  text-align: left;
  width: 100%;
  padding: 4px 0;
}

.box_links_item:first-child {
  font-weight: 700;
  margin-bottom: 4px;
  cursor: default;
}
.box_links_item:not(:first-child):hover {
  color: var(--orange-400);
}

.box_links_item span {
  width: auto;
  padding: 2px 6px;
  height: auto;
  font-size: 12px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  background-color: var(--purple-100);
  color: var(--purple-800);
  border-radius: 4px;
  font-weight: 500;
  line-height: 1;
}

.copyright {
  width: 100%;
  max-width: 1280px;
  min-width: 300px;
  height: auto;
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  justify-content: center;
  margin-top: 24px;
  gap: 16px;
}

.copyright_text {
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.4;
  color: var(--blue-Gray-200);
  text-align: center;
}

.links_acess {
  width: auto;
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-content: center;
  gap: 16px 24px;
}

.links_acess_item {
  font-size: clamp(0.875rem, 0.8281rem + 0.191vw, 1rem);
  font-weight: 400;
  line-height: 1.4;
  color: var(--blue-Gray-200);
  transition: color 0.2s ease-in-out, opacity 0.2s ease-in-out;
  display: inline-flex;
  align-items: center;
  gap: 8px;
  cursor: pointer;
}

.links_acess_item:hover {
  color: var(--orange-400);
  opacity: 0.8;
}

.links_acess_item img {
  filter: brightness(0) saturate(100%) invert(93%) sepia(8%) saturate(238%) hue-rotate(178deg) brightness(93%) contrast(86%);
}
.links_acess_item:hover img {
 filter: brightness(0) saturate(100%) invert(69%) sepia(61%) saturate(5811%) hue-rotate(330deg) brightness(98%) contrast(94%);
}

@media screen and (min-width: 1400px) {
  .content {
    flex-direction: row;
    align-items: flex-start;
    justify-content: space-between;
    width: 90%;
    padding: 48px 0;
  }
  .logo {
    margin-right: 40px;
    align-self: flex-start;
  }
  .box {
    justify-content: space-between;
    flex-wrap: nowrap;
  }
  .copyright {
    width: 90%;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
  }
  .copyright_text {
    text-align: left;
  }
  .links_acess {
    justify-content: flex-end;
  }
}

@media screen and (max-width: 1399px) and (min-width: 841px) {
  .content {
    flex-direction: row;
    align-items: flex-start;
    justify-content: space-between;
    width: 90%;
    padding: 48px 0;
  }
  .logo {
    margin-right: 40px;
    align-self: flex-start;
  }
  .box {
    justify-content: flex-start;
    flex-wrap: wrap;
    gap: 30px 48px;
  }
  .copyright {
    width: 90%;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
  }
  .copyright_text {
    text-align: left;
  }
  .links_acess {
    justify-content: flex-end;
  }
}

@media screen and (max-width: 840px) {
  .content {
    flex-direction: column;
    align-items: flex-start;
    gap: 32px;
    width: 100%;
    padding: 32px 0;
  }
  .logo {
    align-self: flex-start;
    margin-right: 0;
    margin-bottom: 20px;
  }
  .box {
    width: 100%;
    justify-content: flex-start;
    gap: 30px 20px;
  }
  .copyright {
    width: 100%;
    max-width: none;
    flex-direction: column;
    align-items: flex-start;
    text-align: left;
  }
  .copyright_text {
    text-align: left;
  }
  .links_acess {
    width: 100%;
    justify-content: flex-start;
  }
}

@media screen and (max-width: 480px) {
  .content {
    padding: 24px 0;
  }
  .box {
    flex-direction: column;
    align-items: flex-start;
    gap: 24px;
  }
  .box_links {
    width: 100%;
    align-items: flex-start;
    text-align: left;
    gap: 8px;
  }
  .box_links_item {
    padding: 4px 0;
  }
  .copyright_text {
    font-size: 0.8125rem;
  }
  .links_acess_item {
    font-size: 0.8125rem;
  }
}