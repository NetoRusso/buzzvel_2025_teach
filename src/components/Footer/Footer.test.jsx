import { render, screen } from '@testing-library/react';
import Footer from './index';

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props) => {
    // eslint-disable-next-line @next/next/no-img-element, jsx-a11y/alt-text
    return <img {...props} />;
  },
}));

jest.mock('next/link', () => {
  // eslint-disable-next-line react/display-name
  return ({children, href, ...rest}) => {
    // eslint-disable-next-line jsx-a11y/anchor-has-content
    return <a href={href} {...rest}>{children}</a>;
  }
});

describe('Footer Component', () => {
  const RealDate = Date;
  const MOCK_YEAR = 2023;

  beforeEach(() => {
    const mockDate = new Date(MOCK_YEAR, 0, 1);
    // @ts-ignore
    global.Date = class extends RealDate {
      constructor(...args) {
        if (args.length) {
          // @ts-ignore
          return new RealDate(...args);
        }
        return mockDate;
      }
      static now() {
        return mockDate.getTime();
      }
    };
    jest.spyOn(global.Date.prototype, 'getFullYear').mockReturnValue(MOCK_YEAR);
  });

  afterEach(() => {
    global.Date = RealDate;
    jest.restoreAllMocks();
  });

  it('renders the logo', () => {
    render(<Footer />);
    expect(screen.getByAltText('logo')).toBeInTheDocument();
  });

  it('renders all section titles (as p tags)', () => {
    render(<Footer />);
    expect(screen.getByText('Product')).toBeInTheDocument();
    expect(screen.getByText('Solutions')).toBeInTheDocument();
    expect(screen.getByText('Resources')).toBeInTheDocument();
    expect(screen.getByText('Support')).toBeInTheDocument();
    expect(screen.getByText('Company')).toBeInTheDocument();
  });

  it('renders specific link items as text', () => {
    render(<Footer />);
    expect(screen.getByText('Pricing')).toBeInTheDocument();
    expect(screen.getByText('Overview')).toBeInTheDocument();
    expect(screen.getByText((content, element) => element.tagName.toLowerCase() === 'p' && content.startsWith('Accessibility'))).toBeInTheDocument();
    expect(screen.getByText('Brainstorming')).toBeInTheDocument();
    expect(screen.getByText('Help Center')).toBeInTheDocument();
    expect(screen.getByText('Contact Us')).toBeInTheDocument();
    expect(screen.getByText('About')).toBeInTheDocument();
    expect(screen.getByText((content, element) => content.startsWith('Request Demo'))).toBeInTheDocument();
  });

  it('renders the BETA tag for Accessibility', () => {
    render(<Footer />);
    const accessibilityItem = screen.getByText((content, element) => element.tagName.toLowerCase() === 'p' && content.startsWith('Accessibility'));
    expect(accessibilityItem.querySelector('span')).toHaveTextContent('BETA');
  });

  it('renders the "Request Demo" item with its arrow icon', () => {
    render(<Footer />);
    const requestDemoItem = screen.getByText((content, element) => content.startsWith('Request Demo'));
    expect(requestDemoItem).toBeInTheDocument();
    const arrowImage = requestDemoItem.querySelector('img[alt="arrow"]');
    expect(arrowImage).toBeInTheDocument();
  });

  it('renders copyright text with the mocked year', () => {
    render(<Footer />);
    expect(screen.getByText(`uteach Â© ${MOCK_YEAR}. All rights reserved.`)).toBeInTheDocument();
  });

  it('renders accessibility links/items in the copyright section', () => {
    render(<Footer />);
    expect(screen.getByText('Terms')).toBeInTheDocument();
    expect(screen.getByText('Privacy')).toBeInTheDocument();
    expect(screen.getByText('Contact')).toBeInTheDocument(); 
    expect(screen.getByText('EN')).toBeInTheDocument();
    expect(screen.getByAltText('world')).toBeInTheDocument();
    expect(screen.getByText('EUR')).toBeInTheDocument();
    expect(screen.getByAltText('euro')).toBeInTheDocument();
    expect(screen.getByAltText('accessibility')).toBeInTheDocument();
  });

  it('matches snapshot with mocked year', () => {
    const { container } = render(<Footer />);
    expect(container.firstChild).toMatchSnapshot();
  });
});