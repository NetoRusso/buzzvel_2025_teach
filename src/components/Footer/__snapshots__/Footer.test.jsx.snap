// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component matches snapshot with mocked year 1`] = `
<footer
  class="container"
>
  <div
    class="content"
  >
    <img
      alt="logo"
      class="logo"
      height="35"
      loading="lazy"
      src="[object Object]"
      width="173"
    />
    <div
      class="box"
    >
      <div
        class="box_links"
      >
        <p
          class="box_links_item"
        >
          Product
        </p>
        <p
          class="box_links_item"
        >
          Pricing
        </p>
        <p
          class="box_links_item"
        >
          Overview
        </p>
        <p
          class="box_links_item"
        >
          Browse
        </p>
        <p
          class="box_links_item"
        >
          Accessibility 
          <span>
            BETA
          </span>
        </p>
      </div>
      <div
        class="box_links"
      >
        <p
          class="box_links_item"
        >
          Solutions
        </p>
        <p
          class="box_links_item"
        >
          Brainstorming
        </p>
        <p
          class="box_links_item"
        >
          Ideation
        </p>
        <p
          class="box_links_item"
        >
          Wireframing
        </p>
        <p
          class="box_links_item"
        >
          Research
        </p>
      </div>
      <div
        class="box_links"
      >
        <p
          class="box_links_item"
        >
          Resources
        </p>
        <p
          class="box_links_item"
        >
          Help Center
        </p>
        <p
          class="box_links_item"
        >
          Blog
        </p>
        <p
          class="box_links_item"
        >
          Tutorials
        </p>
        <p
          class="box_links_item"
        >
          FAQs
        </p>
      </div>
      <div
        class="box_links"
      >
        <p
          class="box_links_item"
        >
          Support
        </p>
        <p
          class="box_links_item"
        >
          Contact Us
        </p>
        <p
          class="box_links_item"
        >
          Developers
        </p>
        <p
          class="box_links_item"
        >
          Documentation
        </p>
        <p
          class="box_links_item"
        >
          Integrations
        </p>
      </div>
      <div
        class="box_links"
      >
        <p
          class="box_links_item"
        >
          Company
        </p>
        <p
          class="box_links_item"
        >
          About
        </p>
        <p
          class="box_links_item"
        >
          Press
        </p>
        <p
          class="box_links_item"
        >
          Events
        </p>
        <p
          class="box_links_item"
        >
          Request Demo 
          <img
            alt="arrow"
            height="24"
            loading="lazy"
            src="[object Object]"
            width="24"
          />
        </p>
      </div>
    </div>
  </div>
  <div
    class="copyright"
  >
    <p
      class="copyright_text"
    >
      uteach Â© 
      2023
      . All rights reserved.
    </p>
    <div
      class="links_acess"
    >
      <p
        class="links_acess_item"
      >
        Terms
      </p>
      <p
        class="links_acess_item"
      >
        Privacy
      </p>
      <p
        class="links_acess_item"
      >
        Contact
      </p>
      <p
        class="links_acess_item"
      >
        <img
          alt="world"
          height="24"
          loading="lazy"
          src="[object Object]"
          width="24"
        />
        EN
      </p>
      <p
        class="links_acess_item"
      >
        <img
          alt="euro"
          height="24"
          loading="lazy"
          src="[object Object]"
          width="24"
        />
        EUR
      </p>
      <p
        class="links_acess_item"
      >
        <img
          alt="accessibility"
          height="24"
          loading="lazy"
          src="[object Object]"
          width="24"
        />
      </p>
    </div>
  </div>
</footer>
`;
